core/src/test/java/org/infinispan/persistence/support/AsyncStoreTest.java

```diff
@@ -122,11 +139,14 @@ public void testMultiplePutsOnSameKey() throws Exception {
       TestResourceTracker.backgroundTestStarted(this);
       createStore();
 
-      final int number = 1000;
-      String key = "testMultiplePutsOnSameKey-k";
-      String value = "testMultiplePutsOnSameKey-v-";
-      doTestSameKeyPut(number, key, value);
-      doTestSameKeyRemove(key);
+      final int number = 100;
+      final int loops = 10000;
```

Comment:
We need a smaller number of inner and outer loop iterations here, the test timed out when I ran it with trace logging enabled.

I would say extend DummyInMemoryStore to sleep(1) before the write, and use a smaller number of iterations (perhaps 5/100). As it is now, I think the test would fit better in AsyncStoreStressTest.